### PR TITLE
test: e2e localhost test create new channel for existing ica account

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Adding e2e test for localhost interchain accounts channel closing via packet timeout and regaining access to the account by creating a new channel

Steps are as follows:
1. Broadcast `MsgRegisterInterchainAccount` (channel open init).
2. Finish localhost channel handshake by broadcasting msgs for (try, ack, confirm).
3. Send a packet with timeout.
4. Timeout the packet (controller channel end closes), CloseConfirm host channel end.
5. Create new channel by repeating steps 1 and 2.
6. Query the interchain account address and expect the existing balance.
7. SendPacket, RecvPacket, AcknowledgePacket, Verify tokens are transferred.

ref: #3033 

### Commit Message / Changelog Entry

**NA** 

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/interchain_accounts/localhost_test.go [hunk 1]
```diff
@@ -185,6 +185,305 @@ func (s *LocalhostInterchainAccountsTestSuite) TestInterchainAccounts_Localhost(
...
+		}
+
+		msgSendTx := controllertypes.NewMsgSendTx(userAWallet.FormattedAddress(), connectiontypes.LocalhostID, uint64(1), packetData)
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, userAWallet, msgSendTx)
+		s.AssertValidTxResponse(txResp)
+		s.Require().NoError(err)
+
+		events := testsuite.ABCIToSDKEvents(txResp.Events)
+		packet, err = ibctesting.ParsePacketFromEvents(events)
```
Comment (colin-axner) [inline e2e/tests/interchain_accounts/localhost_test.go:326]: :heart:

**Review by colin-axner — APPROVED** [review]
Wow, fantastic work! The code is so clean and beautiful! Was really easy to follow the logic of what was happening

File: e2e/tests/interchain_accounts/localhost_test.go [hunk 2]
```diff
@@ -185,6 +185,305 @@ func (s *LocalhostInterchainAccountsTestSuite) TestInterchainAccounts_Localhost(
...
+	})
+}
+
+func (s *LocalhostInterchainAccountsTestSuite) TestInterchainAccounts_ReopenChannel_Localhost() {
+	t := s.T()
+	ctx := context.TODO()
+
+	// relayer and channel output is discarded, only a single chain is required
+	_, _ = s.SetupChainsRelayerAndChannel(ctx)
+	chainA, _ := s.GetChains()
```
Comment (chatton) [inline e2e/tests/interchain_accounts/localhost_test.go:203]: Maybe we should create an issue to add a second setup function which only spins up a single chain, could be useful for localhost/non-ibc e2e tests.

**Review by chatton — APPROVED** [review]
LGTM left one suggestions for a followup improvement but this looks great! Nice work ❤️

File: e2e/tests/interchain_accounts/localhost_test.go (hunk 2 above)
Comment (damiannolan) [reply e2e/tests/interchain_accounts/localhost_test.go:203]: Yeah, definitely agree!

File: e2e/tests/interchain_accounts/localhost_test.go (hunk 1 above)
Comment (damiannolan) [reply e2e/tests/interchain_accounts/localhost_test.go:326]: Pretty nice how this came to be so valuable here! :D

**Pull request merged**